### PR TITLE
feat(primitives): add error state for SwitchField

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Added `hasError` and `errorMessage` to SwitchField (it already existed on the types, but didn't do anything before)
* Cleaned up some of CheckboxField examples and documentation

#### Description of how you validated changes

CleanShot 2022-09-23 at 10 35 42

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/checkboxfield/examples/CheckboxFieldWithErrorExample.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  return (
+    <form onSubmit={onSubmit}>
+      <CheckboxField
+        label="I agree to the terms and conditions"
+        name="toc"
+        value="yes"
+        checked={checked}
+        hasError={hasError}
+        errorMessage={`Please agree to the terms and conditions`}
```
Comment (zchenwei) [inline docs/src/pages/[platform]/components/checkboxfield/examples/CheckboxFieldWithErrorExample.tsx:26]:
```suggestion
        errorMessage="Please agree to the terms and conditions"
```

File: packages/ui/src/theme/tokens/components/switchField.ts [hunk 2]
```diff
@@ -128,5 +129,9 @@ export const switchfield: SwitchFieldTokens = {
       duration: { value: '{time.short.value}' },
     },
     width: { value: '{space.relative.xl.value}' },
+
+    _error: {
+      backgroundColor: { value: '{colors.border.error.value}' },
```
Comment (zchenwei) [inline packages/ui/src/theme/tokens/components/switchField.ts:134]:
```suggestion
      backgroundColor: { value: '{colors.backgroundColor.error.value}' },
```

**Review by zchenwei — COMMENTED** [review]
Looking good! Left 2 comments

**Review by zchenwei — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM

**Review by 0618 — APPROVED** [review]

**Pull request merged**